### PR TITLE
Remove invalid assertion from parser_flush_cbc

### LINKED ISSUE #4054: Assertion 'last_opcode != PARSER_TO_EXT_OPCODE (CBC_EXT_PUSH_SUPER)' in parser_flush_cbc
###### JerryScript revision
da5b058

###### Build platform
Ubuntu 16.04.6 LTS (Linux 4.15.0-99-generic x86_64)

###### Build steps
```
./tools/build.py --clean --debug --compile-flag=-fsanitize=address \
 --compile-flag=-m32 --compile-flag=-fno-omit-frame-pointer \
 --compile-flag=-fno-common --compile-flag=-g --strip=off \
 --system-allocator=on --logging=on --linker-flag=-fuse-ld=gold \
 --error-messages=on --profile=es2015-subset --lto=off 
```
###### Test case
```
var obj = {
	m() {super.8 = 17;},
};

assert.throws(() => obj.n.call(42), TypeError);

```

###### Output
```
ICE: Assertion 'last_opcode != PARSER_TO_EXT_OPCODE (CBC_EXT_PUSH_SUPER)' failed at /home/jerryscript/jerry-core/parser/js/js-parser-util.c(parser_flush_cbc):156.
Error: ERR_FAILED_INTERNAL_ASSERTION
Aborted (core dumped)

```

Credits: This vulnerability is detected by chong from OWL337.

### PR DESCRIPTION
**Pull request by rerobika**
This patch fixes #4054.

JerryScript-DCO-1.0-Signed-off-by: Robert Fancsik frobert@inf.u-szeged.hu

### CODE REVIEW

File: jerry-core/parser/js/js-lexer.c [hunk 1]
```diff
@@ -1544,7 +1544,7 @@ void
 lexer_next_token (parser_context_t *context_p) /**< context */
 {
   size_t length;
-
+  lexer_token_type_t type = context_p->token.type;
```
Comment (zherczeg) [inline jerry-core/parser/js/js-lexer.c:1547]: `lexer_next_token` should not rdepend on this value

Comment (rerobika) [reply jerry-core/parser/js/js-lexer.c:1547]: So any suggestion?

Comment (zherczeg) [reply jerry-core/parser/js/js-lexer.c:1547]: I think this is correctly parsed as number literal, and the issue should be handled where the number literal is accepted.

**Comment by rerobika** [comment]
@zherczeg Thanks for the review. Indeed, the only problem was an incorrect assertion.

**Pull request merged**